crates/matrix-sdk/src/room/common.rs

```diff
@@ -1108,19 +1112,19 @@ impl<'a> MessagesOptions<'a> {
     /// Creates `MessagesOptions` with the given start token and direction.
     ///
     /// All other parameters will be defaulted.
-    pub fn new(from: &'a str, dir: Direction) -> Self {
+    pub fn new(from: Option<&'a str>, dir: Direction) -> Self {
         Self { from, to: None, dir, limit: uint!(10), filter: RoomEventFilter::default() }
     }
 
     /// Creates `MessagesOptions` with the given start token, and `dir` set to
     /// `Backward`.
-    pub fn backward(from: &'a str) -> Self {
+    pub fn backward(from: Option<&'a str>) -> Self {
```

Comment:
I don't think this is right. Instead, there should be separate `from_start` and `from_end` c'tors like in Ruma.

Reply:
Not sure. The `backwards` method really is about setting the direction, as the name suggest. Whether you provide `from` is optional since v1.3, so I'd reflect that in the method arguments. I'd be also fine with only exposing it on the constructor tho.

Reply:
But, given that ruma doesn't have it either, I'll change it so they match.

Reply:
That's a good point. I have an idea for some API changes to the Ruma equivalent of this, I'll ping you on the PR.